server/id/generator.go

```diff
@@ -41,9 +42,15 @@ func (g randGenerator) ID() ID {
 }
 
 func (g randGenerator) TraceID() trace.TraceID {
-	tid := trace.TraceID{}
-	g.rand.Read(tid[:])
-	return tid
+	var r [16]byte
```

Comment:
@mathnogueira @schoren there is quite an annoying rule by x-ray that requires traces to include a prefix to calculate the creation time, if the trace is older than 30 days the API returns an error, if the prefix is missing x-ray returns an error, I have this code here knowing that is wrong to put it in this place, should we also support a different way of creating traces based on data store? I'm open to ideas

Reply:
Brainstorming: Maybe we should have specializations of this generator (or about the traceid logic) and do something similar with the connectivity PR?